app/src/org/commcare/android/framework/EntitySubnodeDetailFragment.java

```diff
@@ -69,6 +71,17 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
         return rootView;
     }
 
+    private EvaluationContext prepareEvaluationContext(TreeReference childReference) {
+        EvaluationContext sessionContext = asw.getEvaluationContext();
+        EvaluationContext parentDetailContext = new EvaluationContext(sessionContext, childReference);
+        getParentDetail().populateEvaluationContextVariables(parentDetailContext);

```

Comment:
This can through an `XPathTypeMismatchException`, right? Is there a good way to catch this and show it to the user when we aren't in something that extends `CommCareActivity`?